### PR TITLE
HAR-2/Perf/Product-Details-Page

### PR DESCRIPTION
**Pull request by Meekaylah**
# Improved Performance: Fixed Other Products
> Ticket ID-#{HAR-2}

# Changes proposed
> - Showed all Products except selected  product in the Other Products Section
> 
> - Updated "Add to Cart" functionality
>
> - Made UI/UX related changes

### Figma Link
https://www.figma.com/file/vaEXvJixKpCn2lhCdHFzNv/Zuri-Shop?type=design&node-id=107%3A2&mode=design&t=ENO8SvK3kVeptVdd-1

### Hosted Link
https://zuriportfolio-frontend-pw1h.vercel.app/shop/product?id=afe7ae9c-bb42-4d77-9b5a-6a3efe4aac18&shopName=Martinez%20and%20Sons

### Code Checkpoints
- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **main branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots
> image

### CODE REVIEW

**Review by Samadeen — APPROVED** [review]

File: modules/shop/component/otherProductCard.tsx [hunk 1]
```diff
@@ -11,6 +11,9 @@ interface ProductCardProps {
 }
 
 const OtherProductCard: React.FC<ProductCardProps> = ({ product }) => {
+
+  const shopName = router.query.shopName || '';
```
Comment (billmal071) [inline modules/shop/component/otherProductCard.tsx:15]: Why's the shop name allowed to be empty?

File: modules/shop/component/otherProductList.tsx [hunk 2]
```diff
@@ -4,15 +4,27 @@ import ProductCard from '../component/otherProductCard';
...
 interface OtherProductListProps {
   products: Products[];
+  productId: string;
 }
 
-const OtherProductList: React.FC<OtherProductListProps> = ({ products }) => {
+const OtherProductList: React.FC<OtherProductListProps> = ({ products, productId}) => {
+  const filteredArray = products.filter(product => product.id !== productId)
+  console.log('Filtered Array', filteredArray)
+  console.log('ProductId to exclude', productId)
```
Comment (billmal071) [inline modules/shop/component/otherProductList.tsx:13]: Remove the console.logs.

File: modules/shop/productDetails.tsx [hunk 3]
```diff
@@ -60,15 +47,36 @@ export default function ProductDetails() {
...
+          setProduct(response.data);      
+          setShopID(response.data.shop.id) 
         })
         .catch((error) => {
           console.error('Error fetching product details:', error);
           setProduct(null);
         });
     }
   }, [router.query, auth]);
+  console.log('Shop Id', shopID);
```
Comment (billmal071) [inline modules/shop/productDetails.tsx:59]: Remove console.logs.

File: modules/shop/productDetails.tsx [hunk 4]
```diff
@@ -221,20 +229,20 @@ export default function ProductDetails() {
...
+  //   text?.classList.toggle('line-clamp-3');
+  //   text?.classList.toggle('line-clamp-none');
+  //   if (readMoreBtn?.textContent === 'Read more') {
+  //     readMoreBtn.textContent = 'Read less';
+  //     console.log('it reads more');
+  //   } else if (readMoreBtn?.textContent === 'Read less') {
+  //     readMoreBtn.textContent = 'Read more';
+  //     console.log('it reads less');
+  //   }
+  // };
```
Comment (billmal071) [inline modules/shop/productDetails.tsx:245]: Remove unused code.

File: modules/shop/component/otherProductCard.tsx (hunk 1 above)
Comment (Meekaylah) [reply modules/shop/component/otherProductCard.tsx:15]: I eventually fixed that.

File: modules/shop/component/otherProductList.tsx (hunk 2 above)
Comment (Meekaylah) [reply modules/shop/component/otherProductList.tsx:13]: I did. Every single one of them. 😌

File: modules/shop/productDetails.tsx (hunk 4 above)
Comment (Meekaylah) [reply modules/shop/productDetails.tsx:245]: Okay that I didn't do 😅. I'll get to it

File: modules/shop/productDetails.tsx (hunk 3 above)
Comment (Meekaylah) [reply modules/shop/productDetails.tsx:59]: Already done 😌

**Pull request merged**